src/app/list-of-products/list-of-products.component.ts

```diff
@@ -0,0 +1,107 @@
+import {Component, OnDestroy, OnInit, ViewChild} from '@angular/core';
+import {Product} from '../product';
+import {ActivatedRoute, NavigationEnd, Router} from '@angular/router';
+import {PageEvent} from '@angular/material';
+import {MatPaginator} from '@angular/material';
+import {MatDialog} from '@angular/material/dialog';
+import {ProductDetailsDialogComponent} from '../product-details-dialog/product-details-dialog.component';
+
+export interface Sort {
+  value: string;
+  label: string;
+}
+
+const sorts: Sort[] = [
+  {
+    value: 'relevance desc',
+    label: 'Relevance'
+  },
+  {
+    value: 'sort_date desc',
+    label: 'Date of arrivals'
+  },
+  {
+    value: 'current_price asc',
+    label: 'Price low to high'
+  },
+  {
+    value: 'current_price desc',
+    label: 'Price high to low'
+  }
+];
+
+@Component({
+  selector: 'app-list-of-products',
+  templateUrl: './list-of-products.component.html',
+  styleUrls: ['./list-of-products.component.css']
+})
+
+export class ListOfProductsComponent implements OnInit, OnDestroy {
+  pageSize: number;
+  selected: string;
+  products: Product[];
+  navigationSubscription: any;
+  pageEvent: PageEvent;
+  pageIndex: number;
+  count: number;
+  sorts: Sort[] = sorts;
+  private dialogRef: any;
+  currentCategory: string;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+  constructor(private activatedRoute: ActivatedRoute, private router: Router, public dialog: MatDialog) {
+    this.navigationSubscription = this.router.events.subscribe((e: any) => {
+      // If it is a NavigationEnd event re-initalise the component
+      if (e instanceof NavigationEnd) {
+        this.initialiseInvites();
+      }
+    });
```

Comment:
Tricky way to subscribe on resolver changes :) Let's use `this.activatedRoute.data` and remove subscription on all router events.
```typescript
this.activatedRoute.data._subscribe(data => { ... });
```